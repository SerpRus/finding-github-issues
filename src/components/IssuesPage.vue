<template>
  <div class="container">
    <FindingIssuesForm
      :classes="['issues-page__form']"
    />
  </div>
</template>

<script>
import FindingIssuesForm from './FindingIssuesForm.vue';

export default {
  name: 'IssuesPage',
  components: {
    FindingIssuesForm,
  },
};
</script>

<style lang="scss">
.issues-page {
  &__form {
    margin: 50px 0 30px;
  }
}
</style>
