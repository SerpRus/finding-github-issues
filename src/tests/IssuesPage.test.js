import { mount } from '@vue/test-utils';
import IssuesPage from '../components/IssuesPage.vue';

describe('IssuesPage tests', () => {
  const data = [{
    html_url: 'https://github.com/vuejs/vue/pull/12737',
    number: 12737,
    state: 'open',
    created_at: '2022-08-11T09:27:06Z',
    title: 'feat(types): new Vue() improvements (#12730)',
    user: {
      login: 'gulewei',
      html_url: 'https://github.com/gulewei',
    },
  },
  {
    html_url: 'https://github.com/vuejs/vue/issues/12708',
    number: 12708,
    state: 'open',
    created_at: '2022-07-27T20:41:30Z',
    title: '::v-deep is not working',
    user: {
      login: 'handhikadj',
      html_url: 'https://github.com/handhikadj',
    },
  }];

  it('IssuesPage correct issuesData snapshot', () => {
    const wrapper = mount(IssuesPage, {
      data() {
        return {
          issuesData: data,
        };
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('IssuesPage no issuesData snapshot', () => {
    const wrapper = mount(IssuesPage, {
      data() {
        return {
          issuesData: [],
        };
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
