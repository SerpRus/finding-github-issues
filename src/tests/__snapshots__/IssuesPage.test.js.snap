// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssuesPage tests IssuesPage correct issuesData snapshot 1`] = `
"<div class=\\"container\\">
  <form class=\\"finding-form issues-page__form\\">
    <h1 class=\\"finding-form__title\\">Enter repository name</h1>
    <div class=\\"finding-form__content\\"><label for=\\"finding-repository\\" class=\\"finding-form__label\\"><input id=\\"finding-repository\\" class=\\"finding-form__input\\" type=\\"text\\" name=\\"finding-input\\" placeholder=\\"vuejs/vue\\"></label><button class=\\"finding-form__button\\" type=\\"submit\\"> Search </button></div>
  </form>
  <!--v-if-->
  <!--v-if-->
  <ul id=\\"issues-list\\" class=\\"issues-list\\">
    <li class=\\"issue-item\\">
      <div class=\\"issue-item__left-content\\">
        <div class=\\"issue-item__state\\"> State: <span class=\\"issue-item__state-value\\">open</span></div>
        <div class=\\"issue-item__title-wrapper\\"><a class=\\"issue-item__title\\" href=\\"https://github.com/vuejs/vue/pull/12737\\">feat(types): new Vue() improvements (#12730)</a></div>
        <div class=\\"issue-item__info\\">
          <div class=\\"issue-item__number-with-date\\">
            <div class=\\"issue-item__number\\"> #12737</div>
            <div class=\\"issue-item__date\\"> opened on 11 Aug</div>
          </div>
          <div class=\\"issue-item__author\\"> by <a class=\\"issue-item__login\\" href=\\"https://github.com/gulewei\\">gulewei</a></div>
        </div>
      </div>
      <div class=\\"issue-item__right-content\\">
        <!--v-if-->
      </div>
    </li>
    <li class=\\"issue-item\\">
      <div class=\\"issue-item__left-content\\">
        <div class=\\"issue-item__state\\"> State: <span class=\\"issue-item__state-value\\">open</span></div>
        <div class=\\"issue-item__title-wrapper\\"><a class=\\"issue-item__title\\" href=\\"https://github.com/vuejs/vue/issues/12708\\">::v-deep is not working</a></div>
        <div class=\\"issue-item__info\\">
          <div class=\\"issue-item__number-with-date\\">
            <div class=\\"issue-item__number\\"> #12708</div>
            <div class=\\"issue-item__date\\"> opened on 27 Jul</div>
          </div>
          <div class=\\"issue-item__author\\"> by <a class=\\"issue-item__login\\" href=\\"https://github.com/handhikadj\\">handhikadj</a></div>
        </div>
      </div>
      <div class=\\"issue-item__right-content\\">
        <!--v-if-->
      </div>
    </li>
  </ul>
</div>"
`;

exports[`IssuesPage tests IssuesPage no issuesData snapshot 1`] = `
"<div class=\\"container\\">
  <form class=\\"finding-form issues-page__form\\">
    <h1 class=\\"finding-form__title\\">Enter repository name</h1>
    <div class=\\"finding-form__content\\"><label for=\\"finding-repository\\" class=\\"finding-form__label\\"><input id=\\"finding-repository\\" class=\\"finding-form__input\\" type=\\"text\\" name=\\"finding-input\\" placeholder=\\"vuejs/vue\\"></label><button class=\\"finding-form__button\\" type=\\"submit\\"> Search </button></div>
  </form>
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
</div>"
`;
